src/main/java/edu/harvard/iq/dataverse/settings/FileBasedSettingsServiceBean.java

```diff
@@ -0,0 +1,104 @@
+package edu.harvard.iq.dataverse.settings;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.annotation.PostConstruct;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Service responsible for loading and serving application settings
+ * defined in property files. 
+ * 
+ * @author madryk
+ */
+@Startup
+@Singleton
+public class FileBasedSettingsServiceBean {
```

Comment:
I'm not sure if we should still follow ...Bean naming convention.

Reply:
Maybe we should change it to `FileBasedSettingsFetcher`? Or do you have something else in mind?